### PR TITLE
unit preview pane: use overloaded functions to set data

### PR DESCRIPTION
**Pull request by babaissarkar**

### CODE REVIEW

File: src/gui/widgets/unit_preview_pane.hpp [hunk 1]
```diff
@@ -46,11 +46,9 @@ class unit_preview_pane : public container_base
 public:
 	explicit unit_preview_pane(const implementation::builder_unit_preview_pane& builder);
 
-	/** Displays the stats of a specified unit type */
-	void set_displayed_type(const unit_type& type);
-
-	/** Displays the stats of a specific unit */
-	void set_displayed_unit(const unit& u);
+	/** Displays the given data, either from a unit or unit_type */
```
Comment (Vultraz) [inline src/gui/widgets/unit_preview_pane.hpp:49]: PR looks good. One small change, could you revert the change to the doc comments here, like so:
```cpp
/** Displays the stats of a specified unit type */
void set_display_data(const unit_type& type);

/** Displays the stats of a specific unit */
void set_display_data(const unit& u);
```

**Review by Vultraz — CHANGES_REQUESTED** [review]

File: src/gui/widgets/unit_preview_pane.hpp (hunk 1 above)
Comment (babaissarkar) [reply src/gui/widgets/unit_preview_pane.hpp]: Done.
